### PR TITLE
Caching documentaion

### LINKED ISSUE #9: Caching: (ETag)
We should add documentation about caching using `ETag`'s and `If-Match` / `If-None-Match` headers.

### PR DESCRIPTION
**Pull request by krak3n**
This fixes #9

### CODE REVIEW

**Comment by krak3n** [comment]
@SOON-dorks can you all take a look at the caching documentation pwease.

:cat:

**Comment by gregatsoon** [comment]
:+1: ETag ftw!

**Comment by radeklos** [comment]
:+1:

**Comment by jamesjwarren** [comment]
:+1:

File: README.md [hunk 1]
```diff
@@ -848,3 +849,127 @@ func main() {
...
+Accept-Encoding: gzip, deflate
+Connection: keep-alive
+Host: foo.service/123
+If-None-Match: 8594ed77e6c35d97bb7e07c9fa765b15
+```
+
+Once again we get a `304`.
+
+``` http
+HTTP/1.1 304 Ntot Mofified
```
Comment (edoparearyee) [inline README.md]: typo on `Not Modified`

**Comment by edoparearyee** [comment]
other than the above looks good :+1:

**Pull request merged**